Doc/library/array.rst

```diff
@@ -30,9 +30,9 @@ defined:
 +-----------+--------------------+-------------------+-----------------------+-------+
 | ``'H'``   | unsigned short     | int               | 2                     |       |
 +-----------+--------------------+-------------------+-----------------------+-------+
-| ``'i'``   | signed int         | int               | 2                     |       |
+| ``'i'``   | signed int         | int               | 4                     |       |
 +-----------+--------------------+-------------------+-----------------------+-------+
-| ``'I'``   | unsigned int       | int               | 2                     |       |
+| ``'I'``   | unsigned int       | int               | 4                     |       |
```

Comment:
I would expect four bytes on almost every typical computing environment, but I wouldn't expect this to necessarily be true on a 32 bit system. This would depend on the C compiler and system. The C standard only guarantees that a signed or unsigned int should occupy at least 16 bits.

For that reason, I don't support this change.

Reply:
True, @davidlowryduda. OTOH, it could be that CPython required `int` to be at least 32 bits, but I can't find any such requirement in the build system or in any pre-processor guard[^1]. I'm not sure if anyone tried to build and run CPython on a system with 16 bit ints, though.

[^1]: except for the `_datetime` extension module, which explicitly fails to build if `SIZEOF_INT < 4`.

Reply:
@adiaholic, can you please revert the change in `array.rst`?

Reply:
See existing PR:

- #113708